tests/string_test.c

```diff
@@ -0,0 +1,194 @@
+
+    struct aws_byte_buf value = {0};
+    ASSERT_SUCCESS(aws_array_list_get_at(&output, &value, 0), "Retrieving element at 0 failed.");
+
+    char *expected = "testa";
+
+    ASSERT_BIN_ARRAYS_EQUALS(expected, strlen(expected), value.buffer,  value.len, "Value 0 should have been %s", expected);
+
+    aws_array_list_clean_up(&output);
+
+    return 0;
+}
+
+AWS_TEST_CASE(test_char_split_token_not_present, test_char_split_token_not_present_fn)
+
+static int test_char_split_empty_fn(struct aws_allocator *allocator, void *ctx) {
+    const char str_to_split[] = "";
+
+    struct aws_byte_buf to_split = aws_byte_buf_from_literal(str_to_split);
+
+    struct aws_array_list output;
+    ASSERT_SUCCESS(aws_array_list_init_dynamic(&output, allocator, 4, sizeof(struct aws_byte_buf)), "Array list init failed!");
+    ASSERT_SUCCESS(aws_string_split_on_char(&to_split, ';', &output), "String split failed");
+    ASSERT_INT_EQUALS(0, aws_array_list_length(&output), "Split should have returned %d elements", 0);
+
+    aws_array_list_clean_up(&output);
+
+    return 0;
+}
+
+AWS_TEST_CASE(test_char_split_empty, test_char_split_empty_fn)
+
+static int test_char_split_adj_tokens_fn(struct aws_allocator *allocator, void *ctx) {
+    const char str_to_split[] = "testa;;testb;testc";
+
+    struct aws_byte_buf to_split = aws_byte_buf_from_literal(str_to_split);
+
+    struct aws_array_list output;
+    ASSERT_SUCCESS(aws_array_list_init_dynamic(&output, allocator, 4, sizeof(struct aws_byte_buf)), "Array list init failed!");
+    ASSERT_SUCCESS(aws_string_split_on_char(&to_split, ';', &output), "String split failed");
+    ASSERT_INT_EQUALS(4, aws_array_list_length(&output), "Split should have returned %d elements", 4);
+
+    struct aws_byte_buf value = {0};
+    ASSERT_SUCCESS(aws_array_list_get_at(&output, &value, 0), "Retrieving element at 0 failed.");
+
+    char *expected = "testa";
+
+    ASSERT_BIN_ARRAYS_EQUALS(expected, strlen(expected), value.buffer,  value.len, "Value 0 should have been %s", expected);
+
+    ASSERT_SUCCESS(aws_array_list_get_at(&output, &value, 1), "Retrieving element at 1 failed.");
+    expected = "";
+    ASSERT_BIN_ARRAYS_EQUALS(expected, strlen(expected), value.buffer,  value.len, "Value 1 should have been %s", expected);
+
+    ASSERT_SUCCESS(aws_array_list_get_at(&output, &value, 2), "Retrieving element at 2 failed.");
+    expected = "testb";
+    ASSERT_BIN_ARRAYS_EQUALS(expected, strlen(expected), value.buffer,  value.len, "Value 2 should have been %s", expected);
+
+    ASSERT_SUCCESS(aws_array_list_get_at(&output, &value, 3), "Retrieving element at 3 failed.");
+    expected = "testc";
+    ASSERT_BIN_ARRAYS_EQUALS(expected, strlen(expected), value.buffer,  value.len, "Value 3 should have been %s", expected);
+
+    aws_array_list_clean_up(&output);
+
+    return 0;
+}
+
+AWS_TEST_CASE(test_char_split_adj_tokens, test_char_split_adj_tokens_fn)
```

Comment:
Missing edge case: Output array list is too small and can't be expanded.

Reply:
updated.